pageserver/src/consumption_metrics.rs

```diff
@@ -322,3 +352,37 @@ pub async fn collect_metrics_task(
 
     Ok(())
 }
+
+/// Caclculate synthetic size for each active tenant
+pub async fn calculate_synthetic_size_worker(
+    synthetic_size_calculation_interval: Duration,
+) -> anyhow::Result<()> {
+    info!("starting calculate_synthetic_size_worker");
+
+    let mut ticker = tokio::time::interval(synthetic_size_calculation_interval);
+
+    loop {
+        tokio::select! {
+            _ = task_mgr::shutdown_watcher() => {
+                info!("calculate_synthetic_size_worker received cancellation request");
+                return Ok(());
+            },
+        _ = ticker.tick() => {
+
+                let tenants = mgr::list_tenants().await;
+                // iterate through list of Active tenants and collect metrics
+                for (tenant_id, tenant_state) in tenants {
+
+                    if tenant_state != TenantState::Active {
+                        continue;
+                    }
+
+                let tenant = mgr::get_tenant(tenant_id, true).await?;
+                info!("spawn calculate_synthetic_size for tenant {}", tenant_id);
+                tenant.calculate_synthetic_size().await?;
```

Comment:
We probably don't want to use the questionmark here, instead just continue over any inactive tenants and synthetic_size calculation failures.

Reply:
This was done.